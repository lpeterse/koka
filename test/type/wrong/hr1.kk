fun runST( action : forall<h> () -> <st<h>,exn> a ) : exn a  = error("runST")

fun wrong() {
  r = ref(1)
  runST( fun() { !r } )
}

fun main() {
  print("hi")
}