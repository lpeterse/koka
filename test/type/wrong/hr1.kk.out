test/type/wrong/hr1.kk(5,10): error: type is not polymorph enough
 context : runST( fun() { !r } )
 term : fun() { !r }
 inferred type : () -> <st<$h>,exn> int
 is less polymorph as: forall<h1> () -> <st<h1>,exn> int